graylog2-server/src/main/java/org/graylog2/contentpacks/ContentPackInstallationPersistenceService.java

```diff
@@ -126,4 +126,19 @@ public int deleteById(ObjectId id) {
         }
         return installationMetaData;
     }
+
+    public int countInstallationOfEntityById(ModelId contentPackEntityId) {
+        final Set<ContentPackInstallation> allInstallations = loadAll();
+        Optional<Integer> result = allInstallations.stream().map(contentPackInstallation -> {
+            return contentPackInstallation.entities().stream().
+                    filter(nativeEntityDescriptor -> nativeEntityDescriptor.contentPackEntityId()
+                            .equals(contentPackEntityId)).collect(Collectors.toSet()).size();
+        }).reduce((acc, next) -> acc + next );
+
+
+        if (result.isPresent())
```

Comment:
Please use `{}` around if-statements. But I think this whole method can be simplified to the following to move the counting operation into the database:

```java
    /**
     * Returns the number of installations the given content pack entity ID is used in.
     * @param contentPackEntityId the content pack entity ID
     * @return number of installations
     */
    public long countInstallationOfEntityById(ModelId contentPackEntityId) {
        final String field = String.format(Locale.ROOT, "%s.%s", ContentPackInstallation.FIELD_ENTITIES, NativeEntityDescriptor.FIELD_ENTITY_ID);

        return dbCollection.getCount(DBQuery.is(field, contentPackEntityId));
    }
```

This returns the number of installations that use the given entity ID. Before it returned the number of entities in all installations that use the given entity ID.

I think the new behavior is even more correct because in [ContentPackService#uninstallDetails()](https://github.com/Graylog2/graylog2-server/pull/5299/files#diff-dbbfb51fd730e758a0e91273028b7407R206)  we compare this result to `<= 1`. With the old behavior, if a single content pack installation would use the entity ID in several entities, that test would fail but it should not. (same issue [here](https://github.com/Graylog2/graylog2-server/pull/5299/files#diff-dbbfb51fd730e758a0e91273028b7407R261))

Reply:
Since a `content_pack_entity_id` can only be appear once in a installation the error case described above is not valid. Anyhow is the described solution much more efficient than mine. So I adapted the code accordingly.